src/main/java/net/snowflake/client/jdbc/RestRequest.java

```diff
@@ -439,15 +439,15 @@ public static CloseableHttpResponse execute(
     return response;
   }
 
-  static boolean isNonretryableHTTPCode(CloseableHttpResponse response, boolean retryHTTP403) {
+  static boolean isNonRetryableHTTPCode(CloseableHttpResponse response, boolean retryHTTP403) {
     return response != null
         && (response.getStatusLine().getStatusCode() < 500
             || // service unavailable
             response.getStatusLine().getStatusCode() >= 600)
         && // gateway timeout
         response.getStatusLine().getStatusCode() != 408
         && // request timeout
-        (retryHTTP403 || response.getStatusLine().getStatusCode() != 403);
+        (!retryHTTP403 || response.getStatusLine().getStatusCode() != 403);
```

Comment:
This is the fix. Reviewers please carefully review whether this change is correct since it's error prone.

Reply:
This makes sense!